### DISCUSSION TITLE
Why does my budget show data from last week?

### DISCUSSION THREAD

**Discussion by v5rm** [firefly-iii/firefly-iii#5578] [Firefly III] [answered]
![Screenshot (9)_LI](https://user-images.githubusercontent.com/12582751/150337194-e4e05eec-793a-4bb9-ac5c-d8fe6648d9ff.jpg)

Sorry I'm very new to all this. Not sure what I'm doing wrong. The period selected should be this month, but the first two budgets show the budget amount and spendings of last week. The third budget shows that of the entire month. The first two budget items were created with weekly budgets and the third was created with a monthly budget. They are all auto-budgeted with "set fixed amount every period". So I'm totally fine with the first two being weekly budgets and the third one being monthly, but how do I make the first two into budgets for this week? 

Also, I've updated the first budget item to now be a monthly budget. I should have my cron jobs set up correctly. They don't give me errors and the last one fired like two hours ago. However, the first budget still shows up as a weekly budget.

**Comment by JC5** [comment, marked as answer]
This is still a little tricky in the current interface. I'm working on a new layout though.

Here's a few pointers:

Automatic budgets get created on the first day of the period. So if you update your automatic budget settings (you can only set one per budget), it won't apply until the next period starts. For weekly budgets, its monday. For monthly budgets, its the first of the month. So if you change these settings, current budgets will not be affected.

The budget amounts you see in the screenshot can be edited freely. Any new budget amount you set in this view will be monthly. This is because your current view is monthly.

If you go to your Preferences, and set the view to "weekly", you can go back to the budget page and set budget amounts for weeklt periods.

This is a bit complex, which is what I hope to fix in the new layout.

**Reply by v5rm** [reply]
I see. Thanks for the help!
